src/Cake.Common.Tests/Unit/Tools/GitReleaseManager/Label/GitReleaseManagerLabellerTests.cs

```diff
@@ -0,0 +1,220 @@
+            }
+
+            [Fact]
+            public void Should_Throw_If_Owner_Is_Null()
+            {
+                // Given
+                var fixture = new GitReleaseManagerLabellerFixture();
+                fixture.Owner = string.Empty;
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                AssertEx.IsArgumentNullException(result, "owner");
+            }
+
+            [Fact]
+            public void Should_Throw_If_Repository_Is_Null()
+            {
+                // Given
+                var fixture = new GitReleaseManagerLabellerFixture();
+                fixture.Repository = string.Empty;
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                AssertEx.IsArgumentNullException(result, "repository");
+            }
+
+            [Fact]
+            public void Should_Throw_If_Settings_Are_Null()
+            {
+                // Given
+                var fixture = new GitReleaseManagerLabellerFixture();
+                fixture.Settings = null;
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                AssertEx.IsArgumentNullException(result, "settings");
+            }
+
+            [Fact]
+            public void Should_Throw_If_GitReleaseManager_Executable_Was_Not_Found()
+            {
+                // Given
+                var fixture = new GitReleaseManagerLabellerFixture();
+                fixture.GivenDefaultToolDoNotExist();
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                AssertEx.IsCakeException(result, "GitReleaseManager: Could not locate executable.");
+            }
+
+            [Theory]
+            [InlineData("/bin/tools/GitReleaseManager/GitReleaseManager.exe", "/bin/tools/GitReleaseManager/GitReleaseManager.exe")]
+            [InlineData("./tools/GitReleaseManager/GitReleaseManager.exe", "/Working/tools/GitReleaseManager/GitReleaseManager.exe")]
+            public void Should_Use_NuGet_Executable_From_Tool_Path_If_Provided(string toolPath, string expected)
+            {
+                // Given
+                var fixture = new GitReleaseManagerLabellerFixture();
+                fixture.Settings.ToolPath = toolPath;
+                fixture.GivenSettingsToolPathExist();
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal(expected, result.Path.FullPath);
+            }
+
+            [WindowsTheory]
+            [InlineData("C:/GitReleaseManager/GitReleaseManager.exe", "C:/GitReleaseManager/GitReleaseManager.exe")]
+            public void Should_Use_NuGet_Executable_From_Tool_Path_If_Provided_On_Windows(string toolPath, string expected)
```

Comment:
NuGet?

Reply:
Ah, that is a copy paste from the existing tests 😄 I can get that fixed up.

Reply:
Otherwise LGTM, please address and rebase and I'll take a new look.